util/util.go

```diff
@@ -210,6 +210,33 @@ func FindInInterface(iface interface{}, mapKeys ...string) (interface{}, error)
 	return mapped, nil
 }
 
+// CreateTemp Creates a temporary directory so that we may gather results and files before compressing the final
+// artifact. It returns a string representing a filepath, a cleanup function that takes a logger in case of a cleanup error, and an optional error.
+func CreateTemp(parent_directory string) (string, func(hclog.Logger), error) {
+	// default "empty" cleanupfunc
+	emptyCleanup := func(hclog.Logger) {}
+
+	// Create the temporary directory inside the given parent directory
+	tmp, err := os.MkdirTemp(parent_directory, "hcdiag")
+	if err != nil {
+		return "", emptyCleanup, fmt.Errorf("Error creating temp directory, message=%w", err)
+	}
+	tmp, err = filepath.Abs(tmp)
+	if err != nil {
+		return "", emptyCleanup, fmt.Errorf("Error identifying absolute path for temp directory, message=%w", err)
+	}
+
+	// return our temporary directory path, cleanup function, and error
+	return tmp,
+		func(l hclog.Logger) {
```

Comment:
Neat way of getting a logger in here!

Reply:
I debated so many different things for this -- first I had the cleanup function return an error, but then we couldn't really `defer cleanup()` anymore without triggering golangci-lint (unhandled error return). Handling it everywhere made things extremely messy, but I'd still like a chance to report on a cleanup failure, since that could affect a customer. Passing in a logger was my compromise. 🤷